lib/spear.ex

```diff
@@ -1818,4 +1822,304 @@ defmodule Spear do
+  to be _in-flight_ when they are sent to the client. Events remain in-flight
+  until they are `ack/3`-ed, `nack/4`-ed, or until the `:message_timeout`
+  duration is exceeded. If a client `ack/3`s a message, the EventStoreDB
+  will send a new message if any are available.
+
+  The in-flight buffer size is controllable per subscriber through
+  `:buffer_size`. Note that `:message_timeout` applies to each event: if
+  the `:buffer_size` is 5 and five events arrive simultaneously, the client
+  has the duration `:message_timeout` to acknowledge all five events before
+  they are considered stale and are automatically considered nack-ed by
+  the EventStoreDB.
+
+  The `:buffer_size` should align with the consumer's ability to batch
+  process events.
+
+  ## Options
+
+  * `:timeout` - (default: `5_000`ms - 5s) the time to await a subscription
+    confirmation from the EventStoreDB.
+  * `:raw?` - (default: `false`) controls whether events are translated from
+    low-level `Spear.Records.Persistent.read_resp/0` records to
+    `t:Spear.Event.t/0`s. By default `t:Spear.Event.t/0`s are sent to the
+    subscriber.
+  * `:credentials` - (default: `nil`) the credentials to use to connect to the
+    subscription. When not specified, the connection-level credentials are
+    used. Credentials must be a two-tuple `{username, password}`.
+  * `:buffer_size` - (default: `1`) the number of events allowed to be sent
+    to the client at a time. These events are considered in-flight. See the
+    backpressure section above for more information.
+
+  ## Examples
+
+      iex> Spear.connect_to_persistent_subscription(conn, self(), "my_stream", "my_group")
+      iex> flush
+      %Spear.Event{}
+      :ok
+  """
+  @doc since: "0.6.0"
+  @doc api: :persistent
+  @spec connect_to_persistent_subscription(
+          connection :: Spear.Connection.t(),
+          subscriber :: pid() | GenServer.name(),
+          stream_name :: String.t(),
+          group_name :: String.t(),
+          opts :: Keyword.t()
+        ) :: {:ok, subscription :: reference()} | {:error, any()}
+  def connect_to_persistent_subscription(conn, subscriber, stream_name, group_name, opts \\ [])
+
+  def connect_to_persistent_subscription(conn, subscriber, stream_name, group_name, opts)
+      when is_binary(stream_name) and is_binary(group_name) do
+    default_subscribe_opts = [
+      timeout: 5_000,
+      raw?: false,
+      through: &Spear.Reading.decode_read_response/1,
+      credentials: nil,
+      buffer_size: 1
+    ]
+
+    opts =
+      default_subscribe_opts
+      |> Keyword.merge(opts)
+      |> Keyword.merge(stream: stream_name, subscriber: subscriber)
+
+    message =
+      Persistent.read_req(
+        content:
+          {:options,
+           Persistent.read_req_options(
+             stream_identifier: Shared.stream_identifier(streamName: stream_name),
+             group_name: group_name,
+             buffer_size: opts[:buffer_size],
```

Comment:
with EventStore 20+ you control the in-flight buffer in the `connect...` function

in the tcp client this was a setting on the psub itself. this seems like a nice improvement because you might have multiple consumers some which can batch-process more events at once than others based on available resources